aspnetcore/fundamentals/environments.md

```diff
@@ -5,179 +5,135 @@ description: Learn how to control app behavior across multiple environments in A
 monikerRange: '>= aspnetcore-2.1'
 ms.author: riande
 ms.custom: mvc
-ms.date: 12/17/2019
+ms.date: 4/12/2020
 uid: fundamentals/environments
 ---
 # Use multiple environments in ASP.NET Core
 
 ::: moniker range=">= aspnetcore-3.0"
 
-By [Rick Anderson](https://twitter.com/RickAndMSFT)
+By [Rick Anderson](https://twitter.com/RickAndMSFT) and [Kirk Larkin](https://twitter.com/serpent5)
 
 ASP.NET Core configures app behavior based on the runtime environment using an environment variable.
 
-[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/fundamentals/environments/sample) ([how to download](xref:index#how-to-download-a-sample))
+[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/fundamentals/environments/3.1sample) ([how to download](xref:index#how-to-download-a-sample))
 
 ## Environments
 
 ASP.NET Core reads the environment variable `ASPNETCORE_ENVIRONMENT` at app startup and stores the value in [IWebHostEnvironment.EnvironmentName](xref:Microsoft.Extensions.Hosting.IHostEnvironment.EnvironmentName). `ASPNETCORE_ENVIRONMENT` can be set to any value, but three values are provided by the framework:
```

Comment:
Things are a bit different here now. The value used for `EnvironmentName` comes from:

1. `DOTNET_ENVIRONMENT`, which is loaded first.
1. `ASPNETCORE_ENVIRONMENT`, but only if `ConfigureWebHostDefaults` is used. This value overrides `DOTNET_ENVIRONMENT`.

This paragraph also talks about `IWebHostEnvironment`, but the property is declared in `IHostEnvironment`, which `IWebHostEnvironment` implements/extends. This is important for those using the generic host _without_ the web additions, because they'd only care about `IHostEnvironment`.

It looks like both of these observations might be relevant throughout the topic.